csi/controller_server.go

```diff
@@ -578,6 +596,62 @@ func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 	return rsp, err
 }
 
+func (cs *ControllerServer) createCSISnapshotTypeLonghornBackingImage(req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	csiSnapshotName := req.GetName()
+	csiVolumeName := req.GetSourceVolumeId()
+	if len(csiVolumeName) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume name must be provided")
```

Comment:
If creating backing image via snapshot, why volume name is always required? three cases are here.

- create a backing image from a volume
- create a backing image from different sources (like download)
- create a backing image from a staticaly created snapshot content

Here, we only cover the first case? or please clarify this.

Reply:
csi `createSnapsot` will only be triggered when `VolumeSnapshot` source pointing to a volume
```
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  name: test-snapshot-backing
spec:
  volumeSnapshotClassName: longhorn-snapshot-vsc
  source:
    persistentVolumeClaimName: test-vol
```
csi will then trigger the function to create snapshot with the given volume


On the other hand, if `VolumeSnapshot` source pointing to a `VolumeSnapshotContent`, csi won't trigger the `createSnapshot()` function
```
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  name: test-snapshot-on-demand-backing
spec:
  volumeSnapshotClassName: longhorn-snapshot-vsc
  source:
    volumeSnapshotContentName: test-on-demand-backing
```

We can only wait until user create PVC and using the `VolumeSnapshot` and the csi trigger `createVolume()` then we can provision the BackingImage